### ISSUE TITLE
Crash and backtrace

### ISSUE THREAD

**Issue by raphael-proust** [janestreet/jbuilder#148]
When trying to build lwt, I get the error described in https://github.com/ocsigen/lwt/issues/402

More information about my config:

- Everything installed from opam
- 
    ```
    $ jbuilder --version
    1.0+beta10
    ```
- 
    ```
    $ uname -a
    Linux … 4.11.3-1-ARCH #1 SMP PREEMPT Sun May 28 10:40:17 CEST 2017 x86_64 GNU/Linux
    ```

Do you need additional info?

**Comment by diml** [comment]
Looking at the lwt issue, I think what might have happen is that you had a leftover generated file in the source directory from a pre-jbuilder build. This must have been one of the file that was generated by the same rule as `lwt_config.ml`.

When a file is available in the source directory, jbuilder will always prefer to copy it as it rather than use a user rule. However there is an issue when the user rule generate multiple files at once and only some of the targets are available in the source directory. I need to take another look at this.

**Comment by raphael-proust** [comment]
I think this is also what happened.

In this specific case, `jbuilder` not only fails, but also prints `Internal error, please report upstream including the contents of _build/log.` and shows a backtrace. Not sure if, in this specific case, this is acceptable or not.

Additionally, the output says: `Description: "Lwt_config" not found in map <dependency graph in _build/default/src/unix>` which is not very useful. Specifically, it refers to implementation details (`map`) instead of user-level/interface concepts.

**Comment by diml** [comment]
yh, the idea is that such an error shouldn't leek to the user, since essentially there is nothing the user can do to fix it. It's really something that should be fixed in jbuilder. What jbuilder outputs is just to help the jbuilder devs understand what's happening.

**Comment by diml** [comment]
Maybe it should say something like: `internal error, send _build/crash-report upstream`

**Comment by diml** [comment]
Fixed by #218
